### PR TITLE
Sensitive indicators

### PR DESCRIPTION
**Pull request by nathanws**

### CODE REVIEW

File: LEAF_Request_Portal/sources/FormEditor.php [hunk 1]
```diff
@@ -33,44 +33,45 @@ public function __construct($db, $login)
...
-                ':html' => $package['html'],
-                ':htmlPrint' => $package['htmlPrint'],
-                ':required' => $package['required'],
-                ':sort' => isset($package['sort']) ? $package['sort'] : 1, );
-
-            $this->db->prepared_query('INSERT INTO indicators (indicatorID, name, format, description, `default`, parentID, categoryID, html, htmlPrint, required, sort, timeAdded, disabled)
-            								VALUES (null, :name, :format, :description, :default, :parentID, :categoryID, :html, :htmlPrint, :required, :sort, CURRENT_TIMESTAMP, 0)', $vars);
-        }
-        else
+    	    $vars = array(':name' => $package['name'],
```
Comment (dastan-rahimi) [inline LEAF_Request_Portal/sources/FormEditor.php]: Are these variables getting sanitized?

Comment (jeanmarcablevets) [reply LEAF_Request_Portal/sources/FormEditor.php]:
```$this->index['POST']->register('formEditor/newIndicator', function ($args) use ($formEditor) {
            $package = array();
            $package['name'] = XSSHelpers::sanitizeHTML($_POST['name']);
            $package['format'] = strip_tags($_POST['format']);
            $package['description'] = XSSHelpers::sanitizeHTML($_POST['description']);
            $package['default'] = XSSHelpers::sanitizeHTML($_POST['default']);
            $package['parentID'] = (int)$_POST['parentID'];
            $package['categoryID'] = XSSHelpers::xscrub($_POST['categoryID']);
            $package['html'] = $_POST['html'];
            $package['htmlPrint'] = $_POST['htmlPrint'];
            $package['required'] = (int)$_POST['required'];
            $package['is_sensitive'] = $_POST['is_sensitive'];
            $package['sort'] = (int)$_POST['sort'];

            return $formEditor->addIndicator($package);
        });```

In FormEditorController.php, this is the function's only usage.  The `'is_sensitive'`, `'html'`, and `'htmlPrint'` probably need sanitizing.

Comment (nathanws) [reply LEAF_Request_Portal/sources/FormEditor.php]: Yeah, they all need to be.

Comment (dastan-rahimi) [reply LEAF_Request_Portal/sources/FormEditor.php]: i don't think you need sanitize the html field

**Pull request merged**